### PR TITLE
Add controller revision resource

### PR DESCRIPTION
**Pull request by RoniKishner**
##### Short description:
Add controller revision resource to ocp_resources

##### More details:
controller revision is used to control and log the version (changed on edits) of the resource it is attached to

### CODE REVIEW

**Comment by RoniKishner** [comment]
!verified

File: ocp_resources/controller_revision.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from ocp_resources.resource import NamespacedResource
+
+
+class ControllerRevision(NamespacedResource):
```
Comment (myakove) [inline ocp_resources/controller_revision.py:4]: Please include reference to the resource
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/controller-revision-v1/

Please implement under to_dict the required resource data.

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by RoniKishner** [comment]
!verified

File: ocp_resources/controller_revision.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            **kwargs,
+        )
+        self.owner_references = owner_references
+        self.revision_object = revision_object
+
+    def to_dict(self):
+        """
+        Generate intended dict representation of the resource.
+        """
+        self._base_body()
```
Comment (myakove) [inline ocp_resources/controller_revision.py:41]: Please follow other resources example
```
super().to_dict()
if not self.yaml_file:
    <you code here>
```

File: ocp_resources/controller_revision.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        self.res["metadata"]["ownerReference"] = self.owner_references
```
Comment (myakove) [inline ocp_resources/controller_revision.py:42]: You assign `self.owner_references` and `self.revision_object.res`  even if there are `None`

Comment (myakove) [inline ocp_resources/controller_revision.py:42]: Use `self.res.setdefault` to update `self.res`

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by RoniKishner** [comment]
!verified

**Review by myakove — APPROVED** [review]

**Comment by RoniKishner** [comment]
!cherry-pick v4.12

**Review by rnetser — APPROVED** [review]

**Comment by RoniKishner** [comment]
!cherry-pick v4.12

**Pull request merged**